language/tools/move-cli/src/package.rs

```diff
@@ -0,0 +1,227 @@
+        }
+
+        // prepare the output directory
+        let pkg_path = dir_path.join(self.name).join(PKG_BINARY_DIR);
+        dep_dirs.push(pkg_path.clone().into_os_string().into_string().unwrap());
+
+        // if we have processed the package, shortcut the execution
+        if pkg_path.exists() {
+            return Ok(dep_dirs);
+        }
+
+        // dump or build
+        fs::create_dir_all(&pkg_path)?;
+
+        if let Some(prebuilt_dir) = self.binary_dir {
+            let glob = format!("**/*.{}", COMPILED_EXTENSION);
+            for entry in prebuilt_dir.find(&glob).unwrap() {
+                if let DirEntry::File(file) = entry {
+                    let file_path = pkg_path.join(file.path());
+                    let mut fp = File::create(file_path)?;
+                    fp.write_all(file.contents())?;
+                } else {
+                    bail!(
+                        "Unexpected directory name ending with the MV extension: {}",
+                        entry.path().display()
+                    );
+                }
+            }
+        } else {
+            // use a temporary directory to host the sources
+            let tmp_dir = tempdir()?;
+            let mut deps = self.prepare_source_files(tmp_dir.path())?;
+            let tmp_src_dir = tmp_dir
+                .path()
+                .join(self.name)
+                .join(PKG_SOURCE_DIR)
+                .into_os_string()
+                .into_string()
+                .unwrap();
+            // need to exclude self from deps
+            deps.retain(|path| path != &tmp_src_dir);
+
+            // compile
+            let (_, compiled_units) = move_compile(&[tmp_src_dir], &deps, None, None)?;
+
+            // save compiled units to files
+            for unit in compiled_units {
+                let mut data = vec![];
+                let name = match unit {
+                    CompiledUnit::Module { ident, module, .. } => {
+                        module.serialize(&mut data)?;
+                        ident.0.value.name.0.value
+                    }
+                    CompiledUnit::Script { key, script, .. } => {
+                        script.serialize(&mut data)?;
+                        key
+                    }
+                };
+                let file_path = pkg_path.join(name).with_extension(COMPILED_EXTENSION);
+                let mut fp = File::create(file_path)?;
+                fp.write_all(&data)?;
+            }
+        }
+
+        // done
+        Ok(dep_dirs)
+    }
+
+    /// Load modules from the given directory
+    pub fn load_modules(&self, dir_path: &Path) -> Result<Vec<CompiledModule>> {
+        let compiled_modules =
+            find_filenames(
+                &[dir_path.to_str().unwrap().to_owned()],
+                |fpath| match fpath.extension().and_then(|s| s.to_str()) {
+                    None => false,
+                    Some(ext) => ext == COMPILED_EXTENSION,
+                },
```

Comment:
Let's make `move_lang::extension_equals` public which will clean this up a bit

Reply:
Will do